core/crypto/vrf/p256/p256_test.go

```diff
@@ -66,6 +74,60 @@ func TestH2(t *testing.T) {
 	}
 }
 
+func TestFromWrappedKey(t *testing.T) {
+	ctx := context.Background()
+	for _, tc := range []struct {
+		spec   *keyspb.Specification
+		keygen keys.ProtoGenerator
+	}{
+		{
+			spec: &keyspb.Specification{
+				Params: &keyspb.Specification_EcdsaParams{
+					EcdsaParams: &keyspb.Specification_ECDSA{
+						Curve: keyspb.Specification_ECDSA_P256,
+					},
+				},
+			},
+			keygen: func(ctx context.Context, spec *keyspb.Specification) (proto.Message, error) {
+				return der.NewProtoFromSpec(spec)
+			},
+		},
+	} {
+		// Generate VRF key.
+		wrapped, err := tc.keygen(ctx, tc.spec)
+		if err != nil {
+			t.Errorf("NewProtoFromSpec failed: %v", err)
+			continue
+		}
+		vrfPriv, err := NewFromWrappedKey(ctx, wrapped)
+		if err != nil {
+			t.Errorf("NewFromWrappedKey failed: %v", err)
+			continue
+		}
+		vrfPublicPB, err := der.ToPublicProto(vrfPriv.Public())
```

Comment:
Could use `der.MarshalPublicKey(crfPriv.Public())` to avoid unnecessarily wrapping the DER in a proto.

Reply:
done